### ISSUE TITLE
How to include the rg path in the search?

### ISSUE THREAD

**Issue by codeforroad** [ibhagwan/fzf-lua#330]
First of all, this plugin rocks! I much prefer it over telescope!

I might not understand the separation of tasks between fzf and ripgrep very well. Please help me achieve the following in nvim:

When I run `rg --column query | fzf` in a terminal, and then enter for example `orders.js order_items` it will narrow down the results to files that are named `orders.js` (or something similar) that contain the string `order_items`. Which means that fzf looks at the entire result string, path prefix AND the match returned by ripgrep.

When I try the same thing using the fzf-lua plugin inside nvim, however, I can't specify parts of the path prefix. Meaning I get no results for `orders.js` for example.

These are my relevant fzf-lua settings:
```
	fzf_opts = {
		-- options are sent as `<left>=<right>`
		-- set to `false` to remove a flag
		-- set to '' for a non-value flag
		-- for raw args use `fzf_args` instead
		["--ansi"] = "",
		["--tabstop"] = "4",
		["--prompt"] = "> ",
		["--info"] = "default",
		["--height"] = "100%",
		["--layout"] = "reverse",
		["--keep-right"] = "",
		["--hscroll-off"] = "10",
		["--extended"] = "",
	},
```
and for grep: 
```
	grep = {
		prompt = " ",
		input_prompt = " ",
		multiprocess = true, -- run command in a separate process
		git_icons = true, -- show git icons?
		file_icons = true, -- show file icons?
		color_icons = true, -- colorize file|git icons
		-- executed command priority is 'cmd' (if exists)
		-- otherwise auto-detect prioritizes `rg` over `grep`
		-- default options are controlled by 'rg|grep_opts'
		cmd = "rg --vimgrep",
		rg_opts = "--column --line-number --no-heading --color=always --smart-case --max-columns=512 --no-hidden --with-filename",
		grep_opts = "--binary-files=without-match --line-number --recursive --color=auto --perl-regexp",
		-- 'live_grep_glob' options:
		glob_flag = "--iglob .meteor,.git", -- for case sensitive globs use '--glob'
		glob_separator = "%s%-%-", -- query separator pattern (lua): ' --'
	},
```

Thanks!

**Comment by ibhagwan** [comment]
I’m assuming you’re using `grep_project` which doesn’t include the file name as the default setting adds `fzf_opts["--nth"] = '2..'` to the command.

You can use `grep({ search = '' })` or `grep({  fzf_opts = { ["--nth"] = false } })`.

If you wish to have more complex file exclusions I would use `live_grep_glob`, for example, search for `order_items -- *order*` would include all files with `order` in the file name.

**Comment by codeforroad** [comment]
I'm using `live_grep`, actually. I've read up on and added `["--nth"] = "1..",` to my `fzf_opts` but to no avail.
I'd just like to understand how the fzf-lua call differs from the above terminal command so it's not considering the path prefix.

**Comment by ibhagwan** [comment]
Live grep generates a new command with each key press, [read this](https://github.com/ibhagwan/fzf-lua/wiki#grep-vs-live-grep).


For what you need just use `grep` with an empty search string as I commented earlier.

**Comment by ibhagwan** [comment]
Btw, if you wish to see the generated command use `live_grep({ debug = true })` and you’ll understand better what’s happening.

**Comment by ibhagwan** [comment]
>I tried the grep({ search = '' }) but that immediately starts loading results into fzf for every empty string in my project.

It shouldn't be only empty strings, it should load **all lines of your project into fzf**, make sure you don't have a space inside the search term or use `grep_project` instead, I just tried the below and I can search any file and any text as you wanted in the OP:

```lua
-- Both do the same thing, which is load all project lines into
-- fzf **and** include the filename in the search results
:lua require'fzf-lua'.grep({ search = '' })
:lua require'fzf-lua'.grep_project({ fzf_opts = { ["--nth"] = false } })
```

If you wish to see the underying `rg` command, add `debug=true` to the parameters:
```lua
:lua require'fzf-lua'.grep({ search = '', debug=true })
```

I also recommend you try the below so you can understand how the `rg` command is form with `live_grep`:
```lua
:lua require'fzf-lua'.live_grep({ debug=true })
:lua require'fzf-lua'.live_grep_glob({ debug=true })
```

For exmaple, this is how `live_grep_glob` forms it's underying command (`debug=true`):
![image](https://user-images.githubusercontent.com/59988195/151888328-ebd42fad-83b3-42bf-a69f-a10f00ea7859.png)

If you wish to run the exact command in the OP, this is how I would do it:
```lua
:lua require'fzf-lua'.grep({ search = 'query', cmd = 'rg --column', debug=true })
```

Which would result in:
![image](https://user-images.githubusercontent.com/59988195/151888558-86b2e3f5-1761-4d58-a986-65b0b6f3d20b.png)


Hopefully this helps you understand.

**Comment by ibhagwan** [comment]
> I tried the `grep({ search = '' })` but that immediately starts loading results into fzf for every empty string in my project, so for every piece of "empty string" after every letter, which floods my CPU in no time. (a file with one five-letter word in it will render 6 results for example)

Just looked into this issue again and the reason you’re getting 6 results is because you configured `cmd = "rg --vimgrep"`. `vimgrep` flag will show multiple results per line which you do not want.

You also have to understand that `cmd` overrides `rg_opts` (it gets ignored and `cmd` is run instead), delete the `cmd` line and empty search grep will work as expected for you. You also do not need `grep_opts`, since you have `rg` installed it will always be prioritized over `grep` (unless you override it with `cmd =`).

**Comment by codeforroad** [comment]
Alright, awesome! It does exactly what I want, long fzf-able strings consisting of path, number and code line. 

My problem was these two assumptions:
- that whatever was shown in the settings example in the [readme](https://github.com/ibhagwan/fzf-lua#customization) were defaults, so I just copied over the whole example and started adjusting where needed. Because I wanted to be sure I was using `rg` so I uncommented the `grep.cmd` prop because I assumed `rg --vimgrep` was what was called anyway + the `--vimgrep` option seemed like something I wanted.
- that `grep.cmd` and `grep.rg_opts` were actually concatenated and one didn't rule out the other.

Thanks a lot for your time and helping me clear this out! Almost done with my migration from vim to nvim now! Thanks for taking the time to explain all of this, @ibhagwan!

For anyone wanting the same as described in my OP:

```
:lua require'fzf-lua'.grep_project({ fzf_opts = { ["--nth"] = false } })
```
and my `grep` options:
```
		rg_opts = "--column --line-number --no-heading --color=always --smart-case --max-columns=512 --no-hidden",
```
Simple when you know how.
